### PR TITLE
Make material 3 more colorful

### PR DESCRIPTION
**Pull request by newhinton**
image
image
image
image

### CODE REVIEW

**Comment by newhinton** [comment]
@Bnyro Any news? ;)

**Comment by newhinton** [comment]
Sorry, i had to add a few things that i missed before.

I think there is also a bit that could be improved on:

1. Maybe the contrast of the two pickers could be better:
Screenshot_20240117-124149_1

2. The Search-Results are a bit overwhelming. I already opened an issue to prune the results better, but a bit of UI-magic would probably help too.

Screenshot_20240117-124324_1

**Comment by newhinton** [comment]
It might also be nessessary to adjust which specific colors we use from material you.

File: app/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -16,18 +16,42 @@
...
+	<style name="FavoriteCard" parent="Widget.Material3.CardView.Elevated">
 		<item name="android:foreground">?selectableItemBackground</item>
 		<item name="android:clickable">true</item>
-		<item name="android:layout_marginLeft">8dp</item>
-		<item name="android:layout_marginRight">8dp</item>
+		<item name="android:layout_marginStart">8dp</item>
+		<item name="android:layout_marginEnd">8dp</item>
 		<item name="android:layout_marginTop">8dp</item>
+		<item name="cardBackgroundColor">?attr/colorPrimaryContainer</item>
+		<item name="android:textColor">?attr/colorPrimaryContainer</item>
```
[the remark this answers is not in the document]
Comment (Bnyro) [reply app/src/main/res/values/styles.xml:26]: To add some more separation, we could increase the `cardElevation` a bit too make it stand out more, would be my idea.

File: app/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -16,18 +16,42 @@
...
+
+	<style name="ErrorCard" parent="Card">
+		<item name="cardBackgroundColor">?attr/colorErrorContainer</item>
+		<item name="android:textColor">?attr/colorError</item>
+		<item name="android:textColor">?attr/colorError</item>
+		<item name="tint">?attr/colorError</item>
+	</style>
+
+	<style name="NavHeaderCard" parent="Card">
+		<item name="cardBackgroundColor">?attr/colorPrimary</item>
```
Comment (Bnyro) [inline app/src/main/res/values/styles.xml:53]: Maybe we can use a color that looks a bit more similar to the background here? The current color difference looks a bit wild in both the dark and the light theme.

It's generally a really good idea, I'd just prefer a softer color difference.

**Comment by Bnyro** [comment]
For the splash screen, I'd prefer using the Material You `backgroundColor` as background color and the `primaryColor` as color of the Transportr logo, as the `primaryColor` should usually never be used as background (this can also be a bit disturbing to the eyes in dark theme, to see a very light primary color when opening the app in fullscreen)

**Comment by Bnyro** [comment]
TLDR: I generally really like the concept of the changes, I'm just not already 100% d'accord with the colors. Thank you for working on this and looking into my suggestions :)

**Comment by newhinton** [comment]
What do you think of this?

I need to check if tinting the splash icon has side effects somewhere.

By the way, if you find a way to make the status bar transparent in the main map view, please tell me. I made the Trip Detail-one transparent, and it looks just great. It doesnt seem to work for the mapview though.

File: app/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -22,8 +22,8 @@
 		<item name="android:layout_marginStart">8dp</item>
 		<item name="android:layout_marginEnd">8dp</item>
 		<item name="android:layout_marginTop">8dp</item>
-		<item name="cardBackgroundColor">?attr/colorPrimaryContainer</item>
-		<item name="android:textColor">?attr/colorPrimaryContainer</item>
+		<item name="cardBackgroundColor">?attr/colorSurfaceVariant</item>
+		<item name="android:textColor">?attr/colorSurfaceVariant</item>
```
Comment (Bnyro) [inline app/src/main/res/values/styles.xml:26]:
```suggestion
<item name="cardElevation">4dp</item>
```
It's really not common to give cards a background color, the selected one here doesn't match the primary color either.

**Comment by Bnyro** [comment]
For the splash screen and the navigation drawer the changes look good to me :+1:

Unfortunately I don't know either why the status bar color is wrong...

File: app/src/main/res/values/styles.xml (hunk 3 above)
Comment (newhinton) [reply app/src/main/res/values/styles.xml:26]: https://m3.material.io/components/cards/guidelines#3d1946a2-f22d-469a-851c-1f0630ad9cdf

The guidelines state that either style is fine, so if we want to use colored cards we can do so.

No colors set, Elevated style, 4dp elevation:
image

No colors set, Filled style, 4dp elevation:
image

Current Style:
image


Light mode, elevated:
image

Light mode, current:
image



For the dark mode, i prefer the elevated style. The background color of the card just fits better.
For the light mode however, i dont like either. The elevated-style's background is a touch too light and the contrast between the background and the card is very low. The current style also does not look great, the color i used is just not great.

Maybe we need different styles for the light?

Comment (Bnyro) [reply app/src/main/res/values/styles.xml:26]: Yes, that's a good idea :+1:

Anything is fine with me as long as we don't override the card background color

Comment (newhinton) [reply app/src/main/res/values/styles.xml:26]: So elevated for dark mode, and the current style for light mode? This should take care of the contrast, and what i am reading from the color guideline the colorSurfaceVariant is exactly for this kind of situation.

Comment (Bnyro) [reply app/src/main/res/values/styles.xml:26]: Looking at the screenshot you sent, yes :+1:

**Comment by Bnyro** [comment]
Three questions/suggestions left, then good to merge from my side :rocket: Thanks for your efforts!

1. Why did you change the default colors that are being used when Material You is not available? I decided to use the same red primary color as of now, so that users won't be confused by the color change.

2. The following code can be used for the light theme to simplify things, it does exactly the same:
```xml
	<style name="FavoriteCard" parent="Widget.Material3.CardView.Filled">
		<item name="android:foreground">?selectableItemBackground</item>
		<item name="android:clickable">true</item>
		<item name="android:layout_marginStart">8dp</item>
		<item name="android:layout_marginEnd">8dp</item>
		<item name="android:layout_marginTop">8dp</item>
		<item name="cardElevation">3dp</item>
	</style>
```

3. There seems to be some visual glitch on the map view for the home and work card:
map

Comparison to how it should look (and looks in the trip planning fragment):
planner

I've no idea why this happens though ...

**Comment by newhinton** [comment]
To your comments:
1. Because they were blue in your current state. eg.: md_theme_primary = #0061A4, which is a blue :D I used the material 3 designer to generate a pallete for the original red and used that.
2. Oh, yeah true. Will update!
3. Well that is curious, i have the same issue. It is always red. I will investigate.


There are a few things i found that could be improved:
1. The divider-line and fab in the primary card view are ending at the same height, it looks a bit weird
2. The main view toolbar is not transparent
4. The map should be a bit more zoomed out in the trip-view, otherwise parts of the trip might be below the statusbar
5. The startup icon is always white

I will look into 2, 3, and 5 of my remarks.

**Comment by newhinton** [comment]
ahh i know why the color is weird. I dont override it from the pre-31 color file, which contains hardcoded-red coloring.

**Comment by Bnyro** [comment]
1. You're right, I thought I had looked at the color codes, byt obviously the chosen primary colors are mainly red tones. My bad!

**Comment by newhinton** [comment]
@Bnyro Phew, this got bigger than initially expected.

I worked through most of the things i mentioned in my last commend. With the exception of the weirdness of the favorites-list i fixed everything. Im not sure why it is stuck like this.

Another exception is the shadow on the searchbar when setting home/work locations. I did add a radius to the background shadow so that it envelops the entire searchbar, but if one were to look very closely, you can still see a little deviation on the border itself. I think that is fine though.

So i think i am done now!

**Comment by wallalex0** [comment]
Very nice what you did there!

I have found things that I think should be addressed:
- The changelog is dark in light mode (sometimes the text is white on white background).
- The changelog is not closable via back button.
- Adding a bit of padding to the changelog text would be also nice.
- The library list does not use the same design as the contributor list
- Many interactive elements are not accessible and also overlay eachother (beacause of immersiveness, especially in landcape mode), I think these should stay inside usable areas, while keeping the immersiveness
- Add immersiveness (like with the map) to every activity (and also to landscape mode).
- Side menu has strange shadows in landcape mode

This is just a very rough summary, I could provide more details if you need.

**Comment by newhinton** [comment]
> The changelog is dark in light mode (sometimes the text is white on white background).

Hmm, for me it works as intended. Which android did you use?

> The changelog is not closable via back button.

True! But i am not sure if it belongs in this PR or in a ticket in the upstream.

You are right with the library, the "immersiveness" as you call it ;)

> I dont know if this is possible, but it would be nice if the flattened down logo on startup of transportr would be only visible, when these logo types are enabled, and when not instead display the red logo

What do you mean? I dont think the user can choose which type of spash screen is shown.

> Side menu has strange shadows in landcape mode

Can you post screenshots? This and the other landscape-issue might also belong in the upstream as an issue and not here. But in the end @Bnyro has to decide when this PR is read. 

@Bnyro What do you think?

**Comment by wallalex0** [comment]
> Hmm, for me it works as intended. Which android did you use?

Android 13
Screenshot_20240126-193054


> What do you mean? I dont think the user can choose which type of spash screen is shown.

You can change if the app icons in the app launcher are all coloured the same, or use their own coloring. (In background & design settings)
And I meant on basis of this setting, that the splash screen is diffrent.
Screenshot_20240126-191307_1

> Can you post screenshots? This and the other landscape-issue might also belong in the upstream as an issue and not here. But in the end Bnyro has to decide when this PR is read.

Screenshot_20240126-192212

And I forgot to add: The selector for transport type stil has an old design
Screenshot_20240126-193020

**Comment by wallalex0** [comment]
The issue with the changelog appears when switching between dark and light mode (happens after a app restart).

**Comment by Bnyro** [comment]
> Can you post screenshots? This and the other landscape-issue might also belong in the upstream as an issue and not here. But in the end @Bnyro has to decide when this PR is read.

> @Bnyro What do you think?
>
Thanks for involving me into all this, but I must admit that I currently really lack time and motiviation to look into these things :)

I don't mind if you want to take my changes and the ones in this PR to create a new PR at Transportr and close my PR so that you can work on improvements and bug fixes, it's probably easier without me as a middle man. I'll still be there to help if needed though, if you're okay with continuing my work at a new pull request from you at https://github.com/grote/Transportr :)

**Comment by newhinton** [comment]
@Bnyro 

Sure! If you are short on time i am going to make a PR myself. Though generally speaking, i think this is already done and i just wanted your final approval. (Which this was, i guess^^)

You'll probably get pinged about the new pr in a bit :D
